core/src/main/scala/org/bykn/bosatsu/Evaluation.scala

```diff
@@ -258,6 +267,25 @@ case class Evaluation[T](pm: PackageMap.Typed[T], externals: Externals) {
       }
     }
 
+  private[this] val someTrue: Option[Boolean] = Some(true)
+  private[this] val someFalse: Option[Boolean] = Some(false)
```

Comment:
What are the benefits of re-using the same instances rather than just constructing in the function? Maybe worth commenting on since it's a bit surprising.

Reply:
also, you could rename `someTrue` to something like `zeroComesFirst` and `someFalse` to `zeroIsSecond`

Reply:
Allocate once rather than once per call to the method.